src/parser/SparqlParser.cpp

```diff
@@ -329,14 +328,24 @@ void SparqlParser::parseWhere(ParsedQuery* query,
         if (subQ_sel_vars.isAsterisk()) {
           auto subQ_ordVars = subQ_sel_vars.orderedVariablesFromQueryBody();
 
-          for (const auto& subSelectVariables : subQ_ordVars) {
-            ParsedQuery::SelectedVarsOrAsterisk* up_sel_vars =
-                &query->selectClause()._varsOrAsterisk;
-            up_sel_vars->addVariableFromQueryBody(subSelectVariables);
+          for (const auto& subSelectALLVariables : subQ_ordVars) {
+            if (!query->addVariableFromSubQueryBody(query,
```

Comment:
This is only a single variable:
const auto& variable : selectedVariablesFromSubquery
(rename the other stuff accordingly